src/odb/src/db/dbTrackGrid.cpp

```diff
@@ -246,6 +246,38 @@ void dbTrackGrid::getGridPatternY(int i,
   step = grid->_y_step[i];
 }
 
+void dbTrackGrid::addNonUniformGridX(int origin_x,
+                                     int line_count,
+                                     int step,
+                                     int first_last_offset,
+                                     int n_rows)
+{
+  _dbTrackGrid* grid = (_dbTrackGrid*) this;
+  for (int i = 0; i < n_rows; i++) {
+    grid->_x_origin.push_back(origin_x + first_last_offset);
+    grid->_x_count.push_back(line_count);
+    grid->_x_step.push_back(step);
+    origin_x
+        += (2 * first_last_offset) + (line_count * step) + first_last_offset;
+  }
+}
+
+void dbTrackGrid::addNonUniformGridY(int origin_y,
+                                     int line_count,
+                                     int step,
+                                     int first_last_offset,
+                                     int n_rows)
+{
+  _dbTrackGrid* grid = (_dbTrackGrid*) this;
+  for (int i = 0; i < n_rows; i++) {
+    grid->_y_origin.push_back(origin_y);
+    grid->_y_count.push_back(line_count);
+    grid->_y_step.push_back(step);
+    origin_y
+        += (2 * first_last_offset) + (line_count * step) + first_last_offset;
+  }
+}
+
```

Comment:
I don't see any use for these methods in this PR. Also if there is a future use, I would prefer that this for loop to be implemented outside the dbTrackGrid class.

Reply:
It can be done from the outside using the existing methods, and I agree it would be a beter solution, I will start changing it.